oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/DocumentStoreIndexerBase.java

```diff
@@ -78,15 +80,62 @@ protected void setProviders() throws IOException {
         this.indexerProviders = createProviders();
     }
 
+    private static class MongoNodeStateEntryTraverserFactory implements NodeStateEntryTraverserFactory {
+
+        private static final AtomicInteger traverserInstanceCounter = new AtomicInteger(0);
+        final RevisionVector rootRevision;
+        final DocumentNodeStore documentNodeStore;
+        final MongoDocumentStore documentStore;
+        final Logger traversalLogger;
+        final CompositeIndexer indexer;
+        final Closer closer;
+
+        public MongoNodeStateEntryTraverserFactory(RevisionVector rootRevision, DocumentNodeStore documentNodeStore,
```

Comment:
I would probably use the name "DocumentStoreNodeStateEntryTraverserFactory"... I see currently we rely on MongoDocumentStore, but I think in the future this could be changed.

Reply:
I think naming this class generically at present would conflict with the fact that it used MongoDocumentStore currently. IMO generic naming should be done when we extract the specific things out and put the common stuff in generic/abstract/parent class and the specific stuff in concrete/child classes. 
This class is declared private, so that also limits the impact currently. I have also made the constructor private node.